CONTRIBUTING.md

```diff
@@ -108,29 +108,63 @@ For hassle-free code contribution, follow the steps below. NOTE: I mostly work i
+4. On the left-most side panel, click on the icon that looks like four squares with one detaching. A search bar will appear. Use this to install the Docker Extension. It might be suggested, but if not, you can type "Docker" to search for it.
+5. If you are working on a Mac, there is a reported [bug](https://github.com/microsoft/vscode-remote-release/issues/3149) and [workaround](https://github.com/microsoft/vscode-remote-release/issues/3149#issuecomment-646588532) when using the Docker extension with a remote container. To change the Docker path once the Docker extension is installed, click on the `><` icon, type "Settings" and update the `Docker Path` section to be `/usr/local/bin/docker`.
+6. In the top menu bar, click on "Terminal" --> "New Terminal". You should see a terminal open at the bottom and if you type `ls` you should see the files in the Tern repository.
+7. Run `pip3 install tern -e .[dev]` to install Tern and its dependencies.
+8. Test that you can run Tern in your dev container by running: `tern report -i debian:buster`. You should see an SBOM printed to the console for the `debian:buster` container image.
+9. To setup your branch for long-term contributions, follow the commands in step five under the "Setting up a Long Term Development Environment" section.
+10. Create a new branch by running `git checkout -b <newbranch-name>`. You're now ready to start making changes to Tern, which you can do via the Terminal or using GUI editor provided by VS Code. Note that you will need to commit your changes with git using the Terminal command line.
 
-1. Open the tern folder in Visual Studio Code
-1. Click the "Open a Remote Window" button in the bottom left corner (>< icon)
-1. Select "Remote Containers: Reopen in container" from the dropdown
 
-### Before you clone the project
+### Setting up a Long Term Development Environment
 
 You may have already cloned the project and started working on it. If you're reading this after the fact, I would highly recommend you save your work and set up a new development environment in this way.
 
 1. Set up a Python virtual environment that has either the Python 3.6 or Python 3.7 executable. See [here](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments) for instructions on how to set this up using your host machine's Python3. There are also a guide to [managing virtual environments using pipenv](https://docs.python-guide.org/dev/virtualenvs/) but I haven't used it. Once done, you should have a folder created. Change to that folder.
 2. Clone *your fork* of Tern in the virtual environment folder.
 3. Activate your virtual environment `source bin/activate`. NOTE: This specific activate script only works for Bash shells. If you need to activate a Fish Shell or C Shell you should use `source/bin/activate.fish` or `source/bin/activate.csh`, respectively.
```

Comment:
Could we update this section to remove "C Shell" and replace it with ZSH? ZSH is the default terminal on MacOS now, so this may be required if using the default doesn't automatically work.

Reply:
Folks who are using zsh can follow the regular directions to activate the virtualenv I believe so maybe my wording around 'bash shells" needs to change. If folks are using C Shell for whatever reason they will need to use the .csh activation script. I think it's still useful to keep this in here as an FYI?

Reply:
Agreed